src/libraries/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -1528,8 +1492,7 @@ internal static void VarDecFromR4(float input, out DecCalc result)
                 // than 2^93.  So a float with an exponent of -94 could just
                 // barely reach 0.5, but smaller exponents will always round to zero.
                 //
-                const uint SNGBIAS = 126;
-                int exp = (int)(GetExponent(input) - SNGBIAS);
+                int exp = input.Exponent + 1;
```

Comment:
The change is correct in that it matches current behavior, but there isn't anything explaining why its "off by one".

That is `Single.Exponent` and `Double.Exponent` return the exact and correct exponent, while this is the adjusting it to be one higher than expected. I've not sat down to work through "why" the adjustment is needed for `decimal's scale` but this is the type of scenario where there should really be a comment explaining it.

Reply:
The logic looks slightly different for `float`. The original code lacks `&mask`, I'm not sure if the failure is related since VS debugging is somehow broken for me. But I don't expect the logic of `float` to be different from `double`.

Reply:
These helpers should be fixed once this [PR ](https://github.com/dotnet/runtime/pull/72217) gets merged. We tabled it for .NET 7 but now that we are starting to work on .NET 8 it should be revisited.

Reply:
Thanks. Then I think it's better to wait for that PR.